src/scripts/main.js

```diff
@@ -1,3 +1,217 @@
+    const b = second.children[index].innerHTML;
+
+    switch (nameColumn) {
+      case 'Name':
+      case 'Position':
+      case 'Office':
+        return a.localeCompare(b);
+
+      case 'Age':
+      case 'Salary':
+        return toNormalNumber(a) - toNormalNumber(b);
+    }
+  });
+
+  if (directSorting) {
+    sortTable.reverse();
+  }
+
+  tbody.append(...sortTable);
+}
+
+function toNormalNumber(string) {
+  let res = '';
+
+  string.includes('$')
+    ? res = string.slice(1).split(',').join('')
+    : res = string;
+
+  return +res;
+}
+
+tbody.addEventListener('click', (eventOnClick) => {
+  const targetRow = eventOnClick.target.closest('tr');
+
+  targetRow.classList.contains('active')
+    ? targetRow.classList.remove('active')
+    : singleSelectRow(targetRow, 'active', tbody);
+});
+
+function singleSelectRow(target, nameClass, tableBody) {
+  const rows = tableBody.querySelectorAll('tr');
+
+  for (const row of rows) {
+    row.classList.remove(nameClass);
+  }
+
+  return target.classList.toggle(nameClass);
+}
+
+document.querySelector('table').insertAdjacentHTML(
+  'afterend', `
+  <form class="new-employee-form">
+  <label>Name:
+    <input
+      name="name"
+      type="text"
+      data-qa="name"
+      value = ''
+    >
+  </label>
+  <label>Position:
+    <input
+      name="position"
+      type="text"
+      data-qa="position"
+      value = ''
+    >
+  </label>
+  <label>Office:
+    <select name="office" data-qa="office">
+      <option>Tokyo</option>
+      <option>Singapore</option>
+      <option>London</option>
+      <option>New York</option>
+      <option>Edinburgh</option>
+      <option>San Francisco</option>
+    </select>
+  </label>
+  <label>Age:
+    <input
+      name="age"
+      type="number"
+      data-qa="age"
+      value = ''
+    >
+  </label>
+  <label>Salary:
+    <input
+      name="salary"
+      type="number"
+      data-qa="salary"
+      value = ''
+    >
+  </label>
+  <button type="submit">Save to table</button>
+</form>`
+);
+
+const form = document.querySelector('.new-employee-form');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const formValue = new FormData(form);
+  const values = Object.fromEntries(formValue.entries());
+  const tr = document.createElement('tr');
+
+  for (const key in values) {
+    if (!validForm(key, values[key])) {
+      return;
+    }
+
+    tr.insertAdjacentHTML(
+      'beforeend',
+      `<td>${normalizeValueForm(key, values[key])}</td>`
+    );
+  }
+
+  tbody.append(tr);
+  pushNotification('Success!', 'success', 'New employee successfully added');
+  form.reset();
+});
+
+function validForm(key, value) {
+  if (!value) {
+    pushNotification('Error', 'error', 'Fill in all fields');
+
+    return false;
+  }
+
+  if (key === 'name' && value.length < 4) {
+    pushNotification('Error', 'error', 'Name must contain at least 4 letters');
+
+    return false;
+  }
+
+  if (key === 'age' && (+value < 18 || +value > 90)) {
```

Comment:
It's better to set the min and max attributes for the input, this way you will avoid unnecessary checks and it will be clearer to the user